public/app/features/trails/AutomaticMetricQueries/AutoVizPanel.tsx

```diff
@@ -51,7 +51,9 @@ export class AutoVizPanel extends SceneObjectBase<AutoVizPanelState> {
 
   private getVizPanelFor(def: AutoQueryDef) {
     return def
-      .vizBuilder(def)
+      .vizBuilder()
+      .setTitle(def.title)
+      .setUnit(def.unit)
```

Comment:
the title and unit is already part of the def object, so the vizBuilder function should be able to access them (if they close them in a closure)